var/Widget/Archive.php

```diff
@@ -1663,6 +1663,19 @@ private function singleHandle(Query $select, bool &$hasPushed)
         /** 设置归档类型 */
         [$this->archiveType] = explode('_', $this->type);
 
+        /** 特殊处理 revision */
+        if ('revision' == $this->archiveType) {
+            $parent = $this->db->fetchRow(
+                $this->select('type')
+                    ->where(
+                        'table.contents.cid = ?',
+                        $this->parent
+                    )
+                    ->limit(1)
+            );
+            $this->archiveType = $parent['type'];
+        }
```

Comment:
你找的位置非常准确，这里可以用更原生的方法做判断

```suggestion
        if ($this->parameter->preview && $this->type === 'revision') {
            $parent = ContentsFrom::allocWithAlias($this->parent, ['cid' => $this->parent]);
            $this->archiveType = $parent->type;
        } else {
            [$this->archiveType] = explode('_', $this->type);
        }
```